dockerfiles/theia/README.md

```diff
@@ -1,33 +1,101 @@
 # How to Build Theia Image
 
 ## Build image manually using build scripts
-To build Theia image manually You could use build.sh scripts.
-Dockerfile uses multi-stage build feature, so first of all You need build builder image -
-theia-dev:
 
-Example:
+To build all the images you can easily run [build.sh](../../build.sh) script located in the repository root.
 
-```shell
-$ ../theia-dev/build.sh
+CI for checking any pull request in this repository runs this script with `--pr` directive
+```bash
+`build.sh --pr`.
+```
+
+**Note**: `--pr` will build only [limited set](../../docker_image_build.include) of docker images.	
+
+To publish docker images use `--push` directive
+```bash
+`build.sh --push`
+```	
+
+## How to build own Che Theia image with Docker
+
+First you need to build `che-theia-dev` image:	
+
+Run in `dockerfiles/theia-dev` dir:	
+
+```bash	
+./build.sh --build-arg:${GITHUB_TOKEN} --tag:next
 ```
 
-After that, You can build Theia image.
+Then in `dockerfiles/theia` run:
 
-Example:
+```bash	
+./build.sh --build-args:${GITHUB_TOKEN},THEIA_VERSION=master --tag:next --branch:master --git-ref:refs\\/heads\\/master
+```
+
+Where `${GITHUB_TOKEN}` is your GitHub API token, it's used for fetching some vscode libraries that placed on GitHub releases. Without that token build will fail.
+
+This script will build new docker image `eclipse/che-theia-dev:next`. The image will contain Che-Theia based on Theia from master branch.
+
+## Build arguments
+
+General command to build Che-Theia with all possible argiments:
 
 ```bash
-$ ./build.sh --build-args:GITHUB_TOKEN=$GITHUB_TOKEN,THEIA_VERSION=0.3.18 --git-ref:refs\\/heads\\/master
+./build.sh --build-args:${GITHUB_TOKEN},THEIA_VERSION=${THEIA_VERSION} --tag:${IMAGE_TAG} --branch:${THEIA_BRANCH} --git-ref:${THEIA_GIT_REFS} --skip-tests
 ```
 
+### GitHub Token
+
+**`${GITHUB_TOKEN}`**
+
+Some Theia dependencies placed on GitHub repository and fetched from the GitHub by using GitHub API. For that you have to generate [new GitHub API token](https://github.com/settings/tokens) and pass it to the script to have proper build.
+
+Parameter `${GITHUB_TOKEN}` is mandatory.
+
+
 ## Theia version
 
-There's a default Theia version set in the script. This version is then injected in all package.jsons.
-You can override `THEIA_VERSION` by exporting the env before running the script
+**`${THEIA_VERSION}`**
+
+It specifies the Theia version. Build script updates all `package.json` files and uses this value as a version for all `'@theia/*'` packages.
```

Comment:
Not sure to understand when is needed to be changed. Don't forget that this is a documentation for enduser/developers.

Reply:
Do we really need to pass that ? what are the benefits ?

Reply:
Section has been reworked